pkg/pgtune/parallel.go

```diff
@@ -31,13 +31,14 @@ var ParallelKeys = []string{
 
 // ParallelRecommender gives recommendations for ParallelKeys based on system resources.
 type ParallelRecommender struct {
-	cpus int
+	cpus         int
+	MaxBGWorkers int
 }
 
 // NewParallelRecommender returns a ParallelRecommender that recommends based on
 // the given number of cpus.
-func NewParallelRecommender(cpus int) *ParallelRecommender {
-	return &ParallelRecommender{cpus}
+func NewParallelRecommender(cpus int, MaxBGWorkers int) *ParallelRecommender {
+	return &ParallelRecommender{cpus, MaxBGWorkers}
```

Comment:
you messaged in a message about setting boundaries, I think it should be done here where `8` is the min at least.